### PR TITLE
UI: Migrate to ES6 classes

### PR DESCRIPTION
**Pull request by backspace**
This can be reviewed commit-by-commit, though 24eadd2 and e993a71 involve a _lot_ of changes; the passing test suite hopefully provides confidence that things aren’t massively broken. The commit messages have more details but the former commit is the output of the [codemod](https://github.com/ember-codemods/ember-native-class-codemod) with some manual tweaking and the latter shows me attempting to manually replicate the codemod for files it ignored for unknown reasons.

The existence of the `@classic` decorator functions as a list of `FIXME`s that we can work our way through incrementally.

---

You can see the outdated todo list below; I think we should handle Prettier/ESLint conflicts in another branch.

Todos:

* [x] query parameters are broken
* [ ] maybe Prettier and ESLint disagree on indentation of mixin uses 😳
* [x] is Prettier even running at all?
* [x] incorrect ids being used for job queries (incorporate [fix](https://github.com/hashicorp/nomad/compare/f-ui/es6-classes-michael))
* [x] some files were missed by codemod 😞
* [ ] ~the job deployments page is broken (or is it? showing locally when it shouldn’t but not on Netlify)~ (I tried it on `master` and it’s the same so I guess it’s unrelated)
* [x] remove uses of [now-unneeded](https://guides.emberjs.com/release/upgrading/current-edition/native-classes/#toc_properties-and-fields) `Object.freeze` in converted classes

### CODE REVIEW

**Comment by backspace** [comment]
I didn’t mean to make this not a draft, I still need to do some commentary and possible final refinements… I thought there was a way to revert changing to not draft 😳

File: ui/app/adapters/job.js [hunk 1]
```diff
@@ -59,14 +58,15 @@ export default Watchable.extend(WithNamespaceIDs, {
         Job: job.get('_newDefinitionJSON'),
       },
     });
-  },
+  }
 
   update(job) {
     const jobId = job.get('id') || job.get('_idBeforeSaving');
+    // debugger;
```
Comment (DingoEatingFuzz) [inline ui/app/adapters/job.js:65]: I think this was from me. Whoops 😊

**Comment by DingoEatingFuzz** [comment]
Hmmmm, @meirish pointed out that this PR actually adds a bunch of weight to the `nomad-ui.js` file. I don't think it's a blocker, but I definitely think it's worth knowing where it's coming from. Is that really just due to transpiling native classes???

**Comment by DingoEatingFuzz** [comment]
Oh my... Looks like class transpiling for es2015 [is in fact a bloat machine](https://babeljs.io/repl#?browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYGwhgzhAEDKC2YQmgbwL4ChOkjAKgO4D2aWO4U0-AFgE4CmDZ2uVAYsQK53QMAeAFwYA7ACYF6TNJmjQAZsVIBeaACIARmDpqA3LOgRi8BgFkGgmsTEAKAJQy5WdEA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=true&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=7.2.2&externalPlugins=).

**Comment by backspace** [comment]
ugh, that’s sad. I don’t quite understand the relationship of all the settings on that page but it seems like IE 11 is partly to blame 😬

File: ui/app/adapters/job.js (hunk 1 above)
Comment (backspace) [reply ui/app/adapters/job.js]: Thanks for catching this, I tried to look things over before requesting review but the sheer volume is prohibitive 😳

**Pull request merged**